Mathlib/Algebra/FreeMonoid/Basic.lean

```diff
@@ -415,4 +423,37 @@ theorem length_reverse {a : FreeMonoid α} : a.reverse.length = a.length :=
 
 end Reverse
 
+section IsomorphicTypes
+
+variable {α β : Type*}
+
+/-- free monoids over isomorphic types are isomorphic -/
+@[to_additive "if two types are isomorphic, the additive free monoids over those types are
+isomorphic"]
+def congrEquiv (e : α ≃ β) : FreeMonoid α ≃* FreeMonoid β :=
+  MulEquiv.mk' ⟨FreeMonoid.map e.toFun, FreeMonoid.map e.invFun, fun _ => map_symm_apply_map_eq e,
```

Comment:
Why not use `where toFun := FreeMonoid.map e` etc? I think that it's more readable. Also, `MulEquiv.mk'` is the same as `MulEquiv.mk`.

Reply:
indeed it is! thanks